### PR TITLE
Allow optimize to produce a portable compiled file

### PR DESCRIPTION
**Pull request by ibrasho**
Another attempt at: https://github.com/laravel/framework/pull/8776

In response to: https://github.com/laravel/framework/issues/9195

### CODE REVIEW

File: src/Illuminate/Foundation/Console/OptimizeCommand.php [hunk 1]
```diff
@@ -124,6 +125,8 @@ protected function getOptions()
             ['force', null, InputOption::VALUE_NONE, 'Force the compiled class file to be written.'],
 
             ['psr', null, InputOption::VALUE_NONE, 'Do not optimize Composer dump-autoload.'],
+
+            ['portable', null, InputOption::VALUE_NONE, 'Skip files with __DIR__ or __FILE__ to make the cache portable'],
```
Comment (taylorotwell) [inline src/Illuminate/Foundation/Console/OptimizeCommand.php]: Can you add a `.` after the description? :)

**Comment by taylorotwell** [comment]
thanks <3

**Comment by franzliedke** [comment]
Woohoo! Thanks, @ibrasho.

**Comment by dzuelke** [comment]
Can we now make that the default? :p

**Comment by GrahamCampbell** [comment]
No.

**Comment by taylorotwell** [comment]
What files are removed from the optimized file if this is the default? Do we have a count?

**Comment by GrahamCampbell** [comment]
I can find out with that v2 version I had before. Should be really easy to count how many files we skipped.

**Comment by GrahamCampbell** [comment]
Ok, so the only file we seem to be skipping in portable node is "Carbon/Carbon".

**Comment by dzuelke** [comment]
Yup, that's the only one causing trouble.

So you can now weigh whether you want that little micro optimization (Opcache takes care of this problem anyway), or portability out of the box.

**Comment by GrahamCampbell** [comment]
I'm looking into some other issues we have too. Give me a few mins. :)

**Pull request merged**